modules/ROOT/pages/websockets/websocket-connector-client-components.adoc

```diff
@@ -0,0 +1,115 @@
+= About Websocket Client Components
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: connectors, ws, wss, websocket
+
+== Connector Client Components
+
+=== Open Outbound Operation
+
+This operation allows an OUTBOUND socket to establish a connection to a websocket endpoint.
+
+In order to establish a connection a configuration reference must be provided. In addition, parameters like path and uri can be provided. For further reference check connector's technical reference.
+
+The following example opens a connection reusing most of the settings defined in the configuration reference, but others like the uri are overridden.
+
+[source,xml,linenums]
+----
+<flow ​name​=​"connect"​>
+    ​<websocket:open-outbound ​path​=​"/relative/path" uri​=​"ws://mulestocks.com/quotes"​ ​config-ref​=​"ws"​ />
+</flow>
+----
+
+==== Headers and Query Parameters
+
+Default headers and query parameters can be set at the configuration level if they will be used for every request.
+
+However, you can add custom headers or query parameters to your connect operation. These will be sent along with the default headers and parameters in the initial HTTP request to establish the WS connection. They can be set using DataWeave expressions as shown in the example below.
+
+[source,xml,linenums]
+----
+<flow ​name​=​"connect"​>
+    <websocket:open-outbound-socket path="/chatEcho" defaultGroups="#[['AVENGERS']]" config-ref="ws"
+                                        target="connectionAttributes" targetValue="#[attributes]">
+        <websocket:headers><![CDATA[
+                #[
+                    {
+                        chatAlias: 'Captain Marvel',
+                        personalities: 'Vers',
+                        personalities: 'Carol Danvers'
+                    }
+                ]
+            ]]>
+        </websocket:headers>
+        <websocket:query-params><![CDATA[
+                #[
+                    {
+                        theme: 'USA'
+                    }
+                ]
+            ]]>
+        </websocket:query-params>
+    </websocket:open-outbound-socket>
+</flow>
+----
+
+==== Socket Id
```

Comment:
This section is more about what the socket id is than how the operation uses it. This is a cornerstone concept. The The rules that the id must follow, are actually the rules of what an id is.